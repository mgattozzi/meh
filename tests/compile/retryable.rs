use meh::Retry;
use meh::Retryable;

fn main() -> Result<(), String> {
    let _ = test()?;
    Ok(())
}

fn test() -> Retryable<(), String, String> {
    // We can use it in a Retryable
    let whee: Retry<(), String> = Retryable::Ok(())?;
    let Retry::Ok(unit) = whee else {
        unreachable!()
    };

    // This fails to compile as it does not implement try
    let unit = whee?;

    Retryable::Ok(unit)
}
