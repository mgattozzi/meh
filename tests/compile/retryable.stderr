error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> tests/compile/retryable.rs:17:16
   |
17 |     let unit = whee?;
   |                ^^^^^ the `?` operator cannot be applied to type `meh::Retry<(), String>`
   |
   = help: the trait `Try` is not implemented for `meh::Retry<(), String>`
